tools/src/main/java/org/infinispan/tools/store/migrator/StoreIteratorFactory.java

```diff
@@ -13,19 +14,31 @@ class StoreIteratorFactory {
 
    static StoreIterator get(Properties properties) {
       StoreProperties props = new StoreProperties(SOURCE, properties);
-      switch (props.storeType()) {
+      StoreType type = props.storeType();
+      switch (type) {
          case JDBC_BINARY:
          case JDBC_MIXED:
          case JDBC_STRING:
             return new JdbcStoreReader(props);
+      }
+
+      Function<StoreProperties, StoreIterator> factory = fileStoreFactory(props.storeType());
+      if (props.isSegmented())
+         return new SegmentedFileStoreReader(props, factory);
```

Comment:
I am not completely aware, the store could have be segmented and of type RocksDb? If so we could have problems here. I mean, should we restrict to files?

Reply:
It's not possible to use this approach with RocksDB stores as they use the RocksDB ColumnFamily as a basis of segmentation.

I did start to look into SIFS, but I ran into several issues and needed to get the PR up ASAP  for a customer.

Reply:
We should probably throw an exception stating that RocksDB isn't supported with segmentation then.

Reply:
Updated.